### ISSUE TITLE
repacking error

### ISSUE THREAD

**Issue by pintuk** [xiaolu/mkbootimg_tools#31]
I get this error when I try to unpack the working folder -> OUT

mkbootimg_tools$ ./mkboot OUT new-boot.img
mkbootimg from OUT/img_info.
./mkboot: line 145: /home/pintu/PINTU/Y2017/tools/PACK_TOOL/mkbootimg_tools/mkbootfs: No such file or directory
kernel : kernel
ramdisk : new_ramdisk
page size : 2048
kernel size : 14512164
ramdisk size : 20
second_size : 37528
base : 0x10000000
kernel offset : 0x00008000
ramdisk offset : 0x01000000
second_offset :
tags offset : 0x00000100
cmd line :
ramdisk is gzip format.
./mkboot: line 101: /home/pintu/PINTU/Y2017/tools/PACK_TOOL/mkbootimg_tools/mkbootimg: No such file or directory
Make boot.img Error! pls check img_info file.

This is my boot.img image for Amazon fire TV having MTK-8173 chipset

Please help and reply me on email: pintu_agarwal@yahoo.com

**Comment by pintuk** [comment]
Hi,
Any status or suggestion on the above issue?
Please find the boot.img attached.
[Amazon_Fire_TV_MT8173.zip](https://github.com/xiaolu/mkbootimg_tools/files/1227471/Amazon_Fire_TV_MT8173.zip)

**Comment by serankua** [comment]
Install dependencies. Assuming you're on a Debian-based distro:

sudo apt-get install git-core gnupg flex bison gperf build-essential 
zip curl zlib1g-dev gcc-multilib g++-multilib libc6-dev-i386 
lib32ncurses5-dev x11proto-core-dev libx11-dev lib32z-dev ccache 
libgl1-mesa-dev libxml2-utils xsltproc unzip

Sadly there's no mention of this anywhere except under one of the closed issues.

**Comment by TonyStark** [comment]
@ramattos not working
it literally said that mkbootimg file not found. but it is available in directory

**Comment by dylangerdaly** [comment]
You're getting that error because you need 32 libs, specifically libc6-dev-i386.

**Comment by maxime-poulain** [comment]
@dylangerdaly  
Thank you, what the 32 bit lib missing for me !

**Comment by rhrn** [comment]
Solved by docker and ubuntu i386
* from host
```
docker run -it --rm --name mkbootimg_tools i386/ubuntu:latest
```

* from container
```
apt update && apt install cpio gzip git
git clone https://github.com/xiaolu/mkbootimg_tools.git
```

* from host copy `boot.img` to container
```
docker cp boot.img mkbootimg_tools:/mkbootimg_tools/boot.img
```

* from container
```
cd mkbootimg_tools
./mkboot boot.img unpackedboot
# ... modify unpackedboot ...
./mkboot unpackedboot newboot.img
```

* copy `newboot.img` from container to host
```
docker cp mkbootimg_tools:/mkbootimg_tools/newboot.img newboot.img
```

**Comment by JohnRyk** [comment]
@ramattos Thanks! you are nice 👍

**Comment by ghost** [comment]
> Solved by docker and ubuntu i386
> 
> * from host
> 
> ```
> docker run -it --rm --name mkbootimg_tools i386/ubuntu:latest
> ```
> 
> * from container
> 
> ```
> apt update && apt install cpio gzip git
> git clone https://github.com/xiaolu/mkbootimg_tools.git
> ```
> 
> * from host copy `boot.img` to container
> 
> ```
> docker cp boot.img mkbootimg_tools:/mkbootimg_tools/boot.img
> ```
> 
> * from container
> 
> ```
> cd mkbootimg_tools
> ./mkboot boot.img unpackedboot
> # ... modify unpackedboot ...
> ./mkboot unpackedboot newboot.img
> ```
> 
> * copy `newboot.img` from container to host
> 
> ```
> docker cp mkbootimg_tools:/mkbootimg_tools/newboot.img newboot.img
> ```

Thank God you guys found solutions to this I was having to use magiskboot on my Droid which wasn't ideal for my needs. Hopefully one of these methods works out

**Comment by ghost** [comment]
Yep installing the missing libs fixed it!!! Hell yeah guys thank you for this!!!!
